### PR TITLE
#DearInternet - allow video embed only for videos on www.youtube.com

### LINKED ISSUE #5901: #DearInternet - allow YouTube video embed
Allow YouTube video embed so we can show this lovely video on our page: [youtu.be/s7OD5BgFrVM](youtu.be/s7OD5BgFrVM)

/cc @sabrinang @Audreymofo

### PR DESCRIPTION
**Pull request by mmmavis**
Will assign code reviewer once this passes design review.
 
---

Allow video embed only for videos on https://www.youtube.com/embed and update help text

---

Closes #5901 

Since it seems like we only have that [one YouTube video](https://www.youtube.com/watch?v=s7OD5BgFrVM&feature=youtu.be) we wanna show (so far, at least) and that this campaign is only till next Jan, I went with the simplest solution

- If a letter includes a valid YouTube embed video url: embed the video (don't show image even if the letter contains an image)
- If a letter includes a video url: regardless what the url is, display a line: `Video url: [link to video url]`
- If a letter includes an image and doesn't contain a valid YouTube embed video url: show image

See test use cases on https://foundation-s-issue-5901-nsn6o5.herokuapp.com/en/campaigns/dearinternet/

### CODE REVIEW

**Review by sabrinang — COMMENTED** [review]
The use cases and styling look good! 
The only thing that stands out is the text "Video url:" isn't changeable. Thinking ahead perhaps it would be more flexible if the label wasn't there or changeable so for use cases where we'd want to include a URL _(to a website, image source etc.)_ it'd work too and not necessarily limited to video? I'd check with Audrey if we want label copy.

**Comment by mmmavis** [comment]
@sabrinang 

> The only thing that stands out is the text "Video url:" isn't changeable. Thinking ahead perhaps it would be more flexible if the label wasn't there or changeable so for use cases where we'd want to include a URL (to a website, image source etc.) it'd work too and not necessarily limited to video? I'd check with Audrey if we want label copy.

Good point. I'll just remove the label copy so that line stays unlabelled. My availability will be limited next week so I just wanna wrap up all my assigned dev work today.

**Review by sabrinang — APPROVED** [review]
Thanks for removing the label, that will do! 👍

**Comment by mmmavis** [comment]
@Pomax do you have the bandwidth to review this today? We are hoping to get this landed before the holiday shutdown.

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/dear_internet_letter_block.py [hunk 1]
```diff
@@ -27,7 +27,11 @@ class DearInternetLetterBlock(blocks.StructBlock):
     )
 
     video_url = blocks.URLBlock(
-        help_text="Video url to link out to",
+        help_text="Video url to link out to. "
+                  "Note that video embed will show only when the url is a valid YouTube video embed url. "
+                  "Go to your YouTube video and click “Share,” "
+                  "then “Embed,” and then copy and paste the provided URL only. "
+                  "For example: https://www.youtube.com/embed/s7OD5BgFrVM",
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/dear_internet_letter_block.py:34]: 👍

**Review by Pomax — APPROVED** [review]
code looks good

**Pull request merged**